subworkflows/nf-core/splitncigar.nf

```diff
@@ -0,0 +1,56 @@
+//
+// Subworkflow: Run GATK4 SplitNCigarReads with intervals, merge and index BAM file.
+//
+
+include { GATK4_SPLITNCIGARREADS } from '../../modules/nf-core/modules/gatk4/splitncigarreads/main'
+include { SAMTOOLS_MERGE         } from '../../modules/nf-core/modules/samtools/merge/main' //addParams(options: params.samtools_merge_options)
+include { SAMTOOLS_INDEX         } from '../../modules/nf-core/modules/samtools/index/main' //addParams(options: params.samtools_index_options)
+
+workflow SPLITNCIGAR {
+    take:
+    bam             // channel: [ val(meta), [ bam ], [bai] ]
+    fasta           // channel: [ fasta ]
+    fasta_fai       // channel: [ fai ]
+    fasta_dict      // channel: [ dict ]
+    intervals       // channel: [ interval_list]
+
+    main:
+
+    ch_versions       = Channel.empty()
+
+    bam.combine(intervals)
+        .map{ meta, bam, bai, intervals ->
+        new_meta = meta.clone()
+        new_meta.id = meta.id + "_" + intervals.baseName
+        new_meta.sample = meta.id
+        [new_meta, bam, bai, intervals]
+    }.set{bam_interval}
+
+    GATK4_SPLITNCIGARREADS(bam_interval, fasta, fasta_fai, fasta_dict)
+    bam_splitncigar = GATK4_SPLITNCIGARREADS.out.bam
+    ch_versions = ch_versions.mix(GATK4_SPLITNCIGARREADS.out.versions.first())
+
+    bam_splitncigar
+        .map{ meta, bam ->
+            meta.id = meta.sample
```

Comment:
just to be safe I would also run meta.clone() here first

Reply:
Ok.